### PR TITLE
Add a mechanism for expected failures in test262 tests

### PR DESCRIPTION
**Pull request by ptomato**
This was present in the original Python script but not ported as we didn't
have any expected failures at the time.

### CODE REVIEW

File: polyfill/test/parseResults.js [hunk 1]
```diff
@@ -33,21 +49,33 @@ stdin.on('end', function () {
...
   const failedTests = [];
+  const unexpectedPasses = [];
   for (const test of tests) {
     const { result, scenario, file } = test;
-    const message = `${result.pass ? 'PASS' : 'FAIL'} ${file} (${scenario})\n`;
+    const expectedFailure = expectedFailures.has(file);
+    const message = `${PREFIXES[+expectedFailure][+result.pass]} ${file} (${scenario})\n`;
     stdout.write(message);
-    if (!result.pass) failedTests.push({ ...test, message });
+    if (result.pass == expectedFailure) {
```
Comment (Ms2ger) [inline polyfill/test/parseResults.js:58]:
```suggestion
    if (result.pass === expectedFailure) {
```

**Review by Ms2ger — APPROVED** [review]
Assuming you tested it. (Should we have a dummy failing test to make sure this keeps working?)

File: polyfill/test/parseResults.js [hunk 2]
```diff
@@ -1,9 +1,25 @@
...
+  ['FAIL', 'PASS'],
+  ['EXPECTED FAIL', 'UNEXPECTED PASS']
+];
+
 let testOutput = '';
 const now = Date.now();
 
+const expectedFailures = new Set();
+const lines = fs.readFileSync('expected-failures.txt', { encoding: 'utf-8' });
+for (let line of lines.split('\n')) {
```
[the remark this answers is not in the document]
Comment (ptomato) [reply polyfill/test/parseResults.js:16]: `line` is reassigned in the next line, so this is not possible.

**Comment by ptomato** [comment]
I tested it on #1591 which I'll merge directly after this. That should cover having a failing test to make sure this keeps working.

**Pull request merged**